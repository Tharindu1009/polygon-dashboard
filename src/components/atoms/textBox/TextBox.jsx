
import './styles/textBox.scss';
import * as React from 'react';
import CN from "classnames";
import TextField from '@mui/material/TextField';
import InputAdornment from '@mui/material/InputAdornment';
import PersonOutlineIcon from '@mui/icons-material/PersonOutline';
import LockIcon from '@mui/icons-material/Lock';
import SearchIcon from '@mui/icons-material/Search';

function TextBox({ id, size, type, icon, placeholder, onChange, error, helperText }) {

    const TextBoxClasses = CN({
        "textBox textBox__search": type === "search",
    });

    return (
        <div className='textBox'>
            <TextField
                data-testid="textBox"
                fullWidth
                className={TextBoxClasses}
                id={id}
                type={type}
                size={size}
                placeholder={placeholder}
                InputProps={{
                    startAdornment: (
                        <InputAdornment position="start">
                            {icon === "user" && <PersonOutlineIcon data-testid="userIcon" />}
                            {icon === "lock" && <LockIcon data-testid="lockIcon" />}
                            {icon === "search" && <SearchIcon data-testid="searchIcon" />}
                        </InputAdornment>
                    ),
                }}
                onKeyUp={(e) => onChange(e)}
                error={error}
                helperText={helperText}
            />
        </div>
    );
}

export default TextBox;

