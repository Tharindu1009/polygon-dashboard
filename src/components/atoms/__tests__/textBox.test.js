import { render, screen, cleanup, fireEvent } from "@testing-library/react";
import TestRenderer from 'react-test-renderer';
import TextBox from "../textBox/TextBox";

afterEach(() => {
    cleanup();
})
const onClick = jest.fn()

test('should render textBox component', () => {
    render(<TextBox id="username" type="text" icon="user" size="small" placeholder="Username"
        onChange={onClick} error={false} helperText="" />);

    const textBoxElement = screen.getByTestId('textBox');
    expect(textBoxElement).toBeInTheDocument();
})

test('should render PersonOutline Icon when passing user as icon prop', () => {
    render(<TextBox id="username" type="text" icon="user" size="small" placeholder="Username"
        onChange={onClick} error={false} helperText="" />);
    const userIconElement = screen.getByTestId('userIcon');
    expect(userIconElement).toBeInTheDocument();
})

test('should render Lock Icon when passing lock as icon prop', () => {
    render(<TextBox id="username" type="text" icon="lock" size="small" placeholder="Username"
        onChange={onClick} error={false} helperText="" />);
    const lockIconElement = screen.getByTestId('lockIcon');
    expect(lockIconElement).toBeInTheDocument();
})

test('should render Search Icon when passing search as icon prop', () => {
    render(<TextBox id="username" type="text" icon="search" size="small" placeholder="Username"
        onChange={onClick} error={false} helperText="" />);
    const searchIconElement = screen.getByTestId('searchIcon');
    expect(searchIconElement).toBeInTheDocument();
})

test('should display helperText when TextBox has helperText', () => {
    render(<TextBox id="username" type="text" icon="search" size="small" placeholder="Username"
        onChange={onClick} error={false} helperText="Required Username" />);
        const textBox = screen.getByTestId('textBox');
    expect(textBox).toBeInTheDocument();
    expect(textBox).toHaveTextContent('Required Username');
})

test('should execute handler function when onChange textBox', () => {
    render(<TextBox id="username" type="text" icon="search" size="small" placeholder="Username"
        onChange={onClick} error={false} helperText="" />);
    const textBox = screen.getByTestId('textBox');
    fireEvent.keyUp(textBox)
    expect(onClick).toHaveBeenCalledTimes(1);    
})

test('matches textBox snapshot', () => {
    const tree = TestRenderer.create(<TextBox id="username" type="text" icon="user" size="small" placeholder="Username"
        onChange={onClick} error={false} helperText="" />).toJSON();
    expect(tree).toMatchSnapshot();
})