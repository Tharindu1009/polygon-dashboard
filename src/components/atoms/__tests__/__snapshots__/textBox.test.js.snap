// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches textBox snapshot 1`] = `
<div
  className="textBox"
>
  <div
    className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
    data-testid="textBox"
    onKeyUp={[Function]}
  >
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall MuiInputBase-adornedStart css-8j6b76-MuiInputBase-root-MuiOutlinedInput-root"
      onClick={[Function]}
    >
      <div
        className="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeSmall css-ittuaa-MuiInputAdornment-root"
      >
        <span
          className="notranslate"
        >
          ​
        </span>
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="userIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 5.9c1.16 0 2.1.94 2.1 2.1s-.94 2.1-2.1 2.1S9.9 9.16 9.9 8s.94-2.1 2.1-2.1m0 9c2.97 0 6.1 1.46 6.1 2.1v1.1H5.9V17c0-.64 3.13-2.1 6.1-2.1M12 4C9.79 4 8 5.79 8 8s1.79 4 4 4 4-1.79 4-4-1.79-4-4-4zm0 9c-2.67 0-8 1.34-8 4v3h16v-3c0-2.66-5.33-4-8-4z"
          />
        </svg>
      </div>
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall MuiInputBase-inputAdornedStart css-1ua80n0-MuiInputBase-input-MuiOutlinedInput-input"
        disabled={false}
        id="username"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="Username"
        required={false}
        type="text"
      />
      <fieldset
        aria-hidden={true}
        className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="css-ihdtdm"
        >
          <span
            className="notranslate"
          >
            ​
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</div>
`;
